src/testdrive/src/action/kafka/ingest.rs

```diff
@@ -342,9 +346,12 @@ impl Action for IngestAction {
                     producer.send(record, timeout).await
                 });
             }
-        }
-        while let Some(res) = futs.next().await {
-            res.map_err(|(e, _message)| e.to_string_alt())?;
+
+            if iteration % INGEST_BATCH_SIZE == 0 || iteration == (self.repeat - 1) {
+                while let Some(res) = futs.next().await {
```

Comment:
The diff does not do this justice, but the `while` has been moved inside the `iteration` loop.